datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -243,9 +243,7 @@ impl ExecutionPlan for CoalesceBatchesExec {
         child_pushdown_result: ChildPushdownResult,
         _config: &ConfigOptions,
     ) -> Result<FilterPushdownPropagation<Arc<dyn ExecutionPlan>>> {
-        Ok(FilterPushdownPropagation::transparent(
-            child_pushdown_result,
-        ))
+        Ok(FilterPushdownPropagation::all(child_pushdown_result))
```

Comment:
For nodes with 1 child `all` and `any` are equivalent, but I think it's safer to pick `all` by default.